databricks_cli/stack/api.py

```diff
@@ -0,0 +1,246 @@
+
+        if custom_path:
+            custom_path_folder = os.path.dirname(custom_path)
+            if not os.path.exists(custom_path_folder):
+                os.makedirs(custom_path_folder)
+            with open(custom_path, 'w+') as f:
+                click.echo('Storing deploy status metadata to %s' % os.path.abspath(custom_path))
+                json.dump(data, f, indent=2)
+
+    def deploy_job(self, resource_id, job_settings, physical_id=None):
+        job_id = None
+        print("Deploying job %s with settings: \n%s \n" % (resource_id, json.dumps(
+            job_settings, indent=2, separators=(',', ': '))))
+
+        if physical_id:  # job exists
+            job_id = physical_id
+        elif 'name' in job_settings:
+            jobs_same_name = self.jobs_client.get_jobs_by_name(job_settings['name'])
+            if jobs_same_name:
+                creator_name = jobs_same_name[0]['creator_user_name']
+                timestamp = jobs_same_name[0]['created_time'] / MS_SEC
+                date_created = datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
+                click.echo('Warning: Job exists with same name created by %s on %s. Job will '
+                           'be overwritten' % (creator_name, date_created))
+                job_id = jobs_same_name[0]['job_id']
+
+        if job_id:
+            try:
+                # Check if persisted job still exists, otherwise create new job.
+                self.jobs_client.get_job(job_id)
+            except HTTPError:
+                job_id = None
+
+        if job_id:
+            click.echo("Updating Job: %s" % resource_id)
+            self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings})
+            click.echo("Link: %s#job/%s" % (self.host, str(job_id)))
+        else:
+            click.echo("Creating Job: %s" % resource_id)
+            job_id = self.jobs_client.create_job(job_settings)['job_id']
+            click.echo("%s Created with ID %s. Link: %s#job/%s" % (
+                resource_id, str(job_id), self.host, str(job_id)))
+
+        deploy_output = self.jobs_client.get_job(job_id)
+
+        return job_id, deploy_output
+
+    def deploy_resource(self, resource):  # overwrite to be added
+        try:
+            resource_id = resource[RESOURCE_ID]
+            resource_type = resource[RESOURCE_TYPE]
+            resource_properties = resource[RESOURCE_PROPERTIES]
+        except KeyError as e:
+            raise ConfigError("%s doesn't exist in resource config" % str(e))
+
+        # Deployment
+        physical_id = self._get_deployed_resource(resource_id, resource_type)
+
+        if resource_type == JOBS_TYPE:
+            job_id = physical_id['job_id'] if physical_id and 'job_id' in physical_id else None
+            job_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                    job_id)
+            physical_id = {'job_id': job_id}
```

Comment:
Again, `deploy_job` should be responsible for composing the returned value `physical_id`, and`deploy_resource` should not care what is inside